addons/account/models/account_move_line.py

```diff
@@ -1572,9 +1572,8 @@ def write(self, vals):
                 line_to_write -= line
                 continue
 
-            if line.parent_state == 'posted':
-                if any(key in vals for key in ('tax_ids', 'tax_line_id')):
-                    raise UserError(_('You cannot modify the taxes related to a posted journal item, you should reset the journal entry to draft to do so.'))
+            if line.parent_state == 'posted' and any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in vals for field_name in ('tax_ids', 'tax_line_id')):
```

Comment:
```suggestion
            if line.parent_state == 'posted' and any(self.env['account.move']._field_will_change(line, vals, field_name)  for field_name in ('tax_ids', 'tax_line_id')):
```
I guess that's what you meant ?